core/src/cubos/gl/debug_render.cpp

```diff
@@ -0,0 +1,471 @@
+#include <cubos/gl/debug_render.hpp>
+
+#include <list>
+
+#include <glm/gtc/matrix_transform.hpp>
+#include <glm/gtx/transform.hpp>
+#include <glm/gtx/quaternion.hpp>
+
+using namespace cubos;
+using namespace cubos::gl;
+
+RenderDevice* Debug::renderDevice;
+ConstantBuffer Debug::mvpBuffer;
+ShaderBindingPoint Debug::mvpBindingPoint, Debug::colorBindingPoint;
+ShaderPipeline Debug::pipeline;
+RasterState Debug::fillRasterState, Debug::wireframeRasterState;
+Debug::DebugDrawObject Debug::objCube, Debug::objSphere;
+std::list<Debug::DebugDrawRequest> Debug::requests;
+std::mutex Debug::debugDrawMutex;
+
+void Debug::initCube()
+{
+    float verts[] = {// front
+                     -0.5f, -0.5f, 0.5f, 0.5f, -0.5f, 0.5f, 0.5f, 0.5f, 0.5f, -0.5f, 0.5f, 0.5f,
+                     // back
+                     -0.5f, -0.5f, -0.5f, 0.5f, -0.5f, -0.5f, 0.5f, 0.5f, -0.5f, -0.5f, 0.5f, -0.5f};
+
+    auto vb = renderDevice->createVertexBuffer(sizeof(verts), verts, gl::Usage::Static);
+
+    unsigned int indices[] = {// front
+                              0, 1, 2, 2, 3, 0,
+                              // right
+                              1, 5, 6, 6, 2, 1,
+                              // back
+                              7, 6, 5, 5, 4, 7,
+                              // left
+                              4, 0, 3, 3, 7, 4,
+                              // bottom
+                              4, 5, 1, 1, 0, 4,
+                              // top
+                              3, 2, 6, 6, 7, 3};
+
+    objCube.numIndices = sizeof(indices) / sizeof(unsigned int);
+    objCube.ib = renderDevice->createIndexBuffer(sizeof(indices), indices, gl::IndexFormat::UInt, gl::Usage::Static);
+
+    gl::VertexArrayDesc vaDesc;
+    vaDesc.elementCount = 1;
+    vaDesc.elements[0].name = "position";
+    vaDesc.elements[0].type = gl::Type::Float;
+    vaDesc.elements[0].size = 3;
+    vaDesc.elements[0].buffer.index = 0;
+    vaDesc.elements[0].buffer.offset = 0;
+    vaDesc.elements[0].buffer.stride = 3 * sizeof(float);
+    vaDesc.buffers[0] = vb;
+    vaDesc.shaderPipeline = pipeline;
+    objCube.va = renderDevice->createVertexArray(vaDesc);
+}
+
+void Debug::initSphere()
+{
+    float verts[] = {0.0f,
```

Comment:
Don't hardcode stuff like this: its too error prone and looks too ugly. With the cube its alright since its easy to understand.
Instead try to write code which generates a sphere. You can implement something like [this](https://github.com/opencv/opencv/blob/4295ed29fa8c2db2fe255467f822190a8e53b24a/modules/highgui/src/viz3d/viz3d.cpp#L364).